### ISSUE TITLE
Feedback from a mentor perspective

### ISSUE THREAD

**Issue by jleben** [ladieslearningcode/llc-intro-to-browser-extensions#11]
Hi!

I'm not sure a GitHub issue is the best way to provide this feedback, but it seemed the most appropriate of the options I found available. I have just participated as a mentor at the National Learn to Code Day event in Victoria using this content. I'd like to provide my concerns about the content from what I've observed.

I would really like to see honest feedback about the workshop directly from the participants, but my concern was that it was really hard for them to cope with the amount and diversity of information presented in this workshop.

There was 4 different languages that were mentioned and the learners had to interact with directly or indirectly: JSON, JavaScript, HTML and CSS. Matters were complicated further by the fact that some of these language were used indirectly as string values in another language.

There was a lot of boilerplate code that had to be set up before students could see some meaningful effect of their extension.

Working inside a browser extensions required a lot of steps to see a change:
1. Modify code
2. Save code file
3. Reload extension
4. Visit desired page
5. Click on the extension button

This long code-evaluate cycle and the fact that there were up to 3 code files involved (manifest.json, script.js and main.js) also provided a lot of potential points of failure. It made the project hard to debug even for mentors and arguably impossible for learners.

I think the amount of information on the slides left very little time for learners to actually experiment with basic concepts like variables, values, types, operations, functions, loops. In my experience hands-on experimentation is crucial for a first-time coder to get any grasp of the concepts. Instead, the workshop consisted of learners mostly just copy pasting from the slides and modifying a word or two. This could have been improved by letting learners experiment in the browser console. However, in addition to the lack of time for this, in my opinion that also wasn't encouraged enough by the slides (or the instructor).

To sum up, my main concerns are with the cognitive load on the learners preventing them from gaining real understanding, feeling empowered and having fun. This could be improved by reducing the amount and the order of information presented and letting learners experiment more with the basics.

I feel like these workshops aimed at people coding for the first time play a very crucial role in attracting or deterring people from the world of coding. In my experience, cognitive load has a big impact on the outcome, and I wish it was considered more carefully. I don't know if Canada Learning Code has a set of content-design principles to ensure adequate cognitive load, but I think it should.

Thank you for considering this feedback and for all your hard work!

I woud be happy to help - not only with mentoring but also with content creation. However, I feel like the concerns I am raising here require more discussion and collaboration than a GitHub pull request would allow.

**Comment by KoltonG** [comment]
@jleben I totally agree! Myself, being a lead instructor, struggled with the amount of new information and concepts that the learners had to understand. 

I would love to participate as well in this conversation and even see this course load have some of the core concepts of the Intro to Javascript course too.

**Comment by sdaponte** [comment]
Hey @jleben and @KoltonG,

Thank you so much for your insights here! I was thinking of making a couple changes to address these issues you have both mentioned above:
- We can provide the learners with the complete JSON file, so they can modify the values of these instead of having to add to the file on their own or by copying & pasting. That way they're learning through tinkering
- Allocating some time at the end (last 20-30 mins) for customization time/work session, OR blocks of time could be staggered throughout the class, so they can get more of that hands-on experimentation. In my experience when shorter bits of time are staggered throughout, it's used more for socializing (or networking with mentors) than on applying the content, but I'm curious if you have had different experiences with that yourself.

Many thanks,
Stacie
